api/app/references/api.py

```diff
@@ -105,3 +114,41 @@ def get(self):
 Kind regards,
 The {event_description} team.
 """
+
+
+class ReferenceAPI(ReferenceMixin, restful.Resource):
+
+    @marshal_with(reference_fields)
+    @auth_required
+    def get(self):
+        LOGGER.debug('Received get request for reference')
+        user = user_repository.get_by_id(g.current_user['id'])
+        if not (user or user.is_admin):
+            return FORBIDDEN
+
+        args = self.get_req_parser.parse_args()
+        response = response_repository.get_by_id_and_user_id(args['response_id'], user.id)
```

Comment:
Not sure why the response repository has a user_id parameter when the response_id is enough to unique identify the record... Can you add another method that's just get_by_id and use that?

Reply:
Gotcha, I agree that is totally not necessary.

Reply:
@avishkar58 @bamtak Is it not to prevent a user from looking up some other users response?

Reply:
I am leaving this part @avishkar58 what do you think.

Reply:
Sure, think that's fine for now, thanks @bamtak !